src/liballoc/collections/linked_list.rs

```diff
@@ -986,6 +1080,321 @@ impl<T> IterMut<'_, T> {
+    pub fn current(&mut self) -> Option<&mut T> {
+        unsafe { self.current.map(|current| &mut (*current.as_ptr()).element) }
+    }
+
+    /// Get the next element
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn peek_next(&mut self) -> Option<&mut T> {
+        unsafe {
+            let next = match self.current {
+                None => self.list.head,
+                Some(current) => current.as_ref().next,
+            };
+            next.map(|next| &mut (*next.as_ptr()).element)
+        }
+    }
+
+    /// Get the previous element
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn peek_prev(&mut self) -> Option<&mut T> {
+        unsafe {
+            let prev = match self.current {
+                None => self.list.tail,
+                Some(current) => current.as_ref().prev,
+            };
+            prev.map(|prev| &mut (*prev.as_ptr()).element)
+        }
+    }
+
+    /// Get an immutable cursor at the current element
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn as_cursor<'cm>(&'cm self) -> Cursor<'cm, T> {
+        Cursor { list: self.list, current: self.current, index: self.index }
+    }
+}
+
+// Now the list editing operations
+
+impl<'a, T> CursorMut<'a, T> {
+    /// Insert `item` after the cursor
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn insert_after(&mut self, item: T) {
+        unsafe {
+            let spliced_node = Box::into_raw_non_null(Box::new(Node::new(item)));
+            let (node, node_next) = match self.current {
+                None => (None, self.list.head),
+                Some(node) => {
+                    let node_next = node.as_ref().next;
+                    (Some(node), node_next)
+                }
+            };
+            self.list.splice_nodes(node, node_next, spliced_node, spliced_node, 1);
+            if self.current.is_none() {
+                // The ghost element's index has increased by 1.
+                self.index += 1;
+            }
+        }
+    }
+
+    /// Insert `item` before the cursor
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn insert_before(&mut self, item: T) {
+        unsafe {
+            let spliced_node = Box::into_raw_non_null(Box::new(Node::new(item)));
+            let (node_prev, node) = match self.current {
+                None => (self.list.tail, None),
+                Some(node) => {
+                    let node_prev = node.as_ref().prev;
+                    (node_prev, Some(node))
+                }
+            };
+            self.list.splice_nodes(node_prev, node, spliced_node, spliced_node, 1);
+            self.index += 1;
+        }
+    }
+
+    /// Remove and return the current item, moving the cursor to the next item
+    #[unstable(feature = "linked_list_cursors", issue = "58533")]
+    pub fn remove(&mut self) -> Option<T> {
```

Comment:
As per https://github.com/rust-lang/rfcs/pull/2847, this should be renamed to `remove_current`.

Reply:
done